### PR TITLE
More advanced options for the Filesystem dialog

### PR DESCRIPTION
**Pull request by d2dyno1**
**Resolved / Related Issues**
- Improves #4411 

**Details of Changes**
Adds more advanced dialog and options for resolving conflicts. Now user can select the option they want per file basis. 
Also UI got some refresh too!

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

### CODE REVIEW

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 1]
```diff
@@ -118,34 +149,39 @@
...
                                     </Setter>
                                 </Style>
-                            </GridView.ItemContainerStyle>
-                            <GridView.ItemTemplate>
+                            </ListView.ItemContainerStyle>
+                            <ListView.ItemTemplate>
                                 <DataTemplate>
                                     <Border>
-                                        <Grid ColumnSpacing="12">
+                                        <Grid ColumnSpacing="12" RowSpacing="15">
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:156]:
```suggestion
                                        <Grid ColumnSpacing="12" RowSpacing="16">
```
Keep in mind, these numbers should be divisible by 4.

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

File: Files/Dialogs/FilesystemOperationDialog.xaml (hunk 1 above)
Comment (d2dyno1) [reply Files/Dialogs/FilesystemOperationDialog.xaml:156]: Aaaaaaaaaaaa

This is still wip 😉

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 2]
```diff
@@ -45,7 +50,7 @@
         </ResourceDictionary>
     </ContentDialog.Resources>
 
-    <Border MaxWidth="500">
+    <Border Width="500">
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:53]: Is it possible to use a smaller width when the user is just deleting items? 500 is a little wide for that scenario.

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 3]
```diff
@@ -91,18 +95,44 @@
...
+                                Command="{x:Bind ViewModel.OptionGenerateNewNameCommand, Mode=OneWay}"/>
+                            <Button 
+                                HorizontalAlignment="Center"
+                                Content="Replace existing"
+                                Command="{x:Bind ViewModel.OptionReplaceExistingCommand, Mode=OneWay}"/>
+                            <Button 
+                                HorizontalAlignment="Center"
+                                Content="Skip"
+                                Command="{x:Bind ViewModel.OptionSkipCommand, Mode=OneWay}"/>
+                        </StackPanel>
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:119]: Can you add strings for these buttons?

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 4]
```diff
@@ -118,34 +148,40 @@
...
                                     </Setter>
                                 </Style>
-                            </GridView.ItemContainerStyle>
-                            <GridView.ItemTemplate>
+                            </ListView.ItemContainerStyle>
+                            <ListView.ItemTemplate>
                                 <DataTemplate>
                                     <Border>
-                                        <Grid ColumnSpacing="12">
+                                        <Grid ColumnSpacing="12" RowSpacing="4" Padding="2 10 2 10">
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:155]:
```suggestion
                                        <Grid ColumnSpacing="12" RowSpacing="4" Padding="2, 10, 2, 10">
```

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 5]
```diff
@@ -159,50 +195,75 @@
...
-                                            <TextBlock
-                                                Grid.Column="1"
-                                                HorizontalAlignment="Center"
-                                                VerticalAlignment="Center"
-                                                Text="{Binding SourcePath}" />
+                                            <!-- Filenames -->
+                                            <StackPanel Orientation="Horizontal" Spacing="8" Grid.Column="1" Grid.Row="0">
+                                                <!-- Source Filename -->
+                                                <TextBlock 
+                                                    FontSize="13"
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:202]:
```suggestion
                                                    FontSize="12"
```

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 6]
```diff
@@ -159,50 +195,75 @@
...
 
-                                            <!--  Destination file path  -->
-                                            <TextBlock
-                                                Grid.Column="3"
-                                                HorizontalAlignment="Center"
-                                                VerticalAlignment="Center"
-                                                Text="{Binding DestinationPath}" />
+                                                <!-- Destination Filename -->
+                                                <TextBlock 
+                                                    FontSize="13"
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:218]:
```suggestion
                                                    FontSize="12"
```

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 7]
```diff
@@ -159,50 +195,75 @@
...
+                                                    Text="{Binding DestinationDirectoryDisplayName}" />
+                                            </StackPanel>
+                                            
+                                            <!-- Combobox with options -->
+                                            <ComboBox Visibility="{Binding DestinationLocationVisibility}" HorizontalAlignment="Right" Grid.Column="2" Grid.RowSpan="2" SelectedIndex="{Binding ConflictResolveOptionIndex, Mode=TwoWay}">
+                                                <ComboBox.Items>
+                                                    <ComboBoxItem Content="Generate new name"/>
+                                                    <ComboBoxItem Content="Replace"/>
+                                                    <ComboBoxItem Content="Skip"/>
+                                                </ComboBox.Items>
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:260]: Can you add strings for these items?

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 8]
```diff
@@ -159,50 +195,75 @@
...
+                                                <TextBlock
+                                                    HorizontalAlignment="Center"
+                                                    VerticalAlignment="Center"
+                                                    Foreground="Gray"
+                                                    Visibility="{Binding DestinationLocationVisibility}"
+                                                    Text="{Binding DestinationDirectoryDisplayName}" />
+                                            </StackPanel>
+                                            
+                                            <!-- Combobox with options -->
+                                            <ComboBox Visibility="{Binding DestinationLocationVisibility}" HorizontalAlignment="Right" Grid.Column="2" Grid.RowSpan="2" SelectedIndex="{Binding ConflictResolveOptionIndex, Mode=TwoWay}">
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:255]: Can you use a split button?

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 9]
```diff
@@ -38,174 +45,262 @@
...
+
+                                        <!--  Split with options  -->
+                                        <muxc:SplitButton
+                                            x:Name="Split"
+                                            Grid.RowSpan="2"
+                                            Grid.Column="2"
+                                            HorizontalAlignment="Right"
+                                            VerticalAlignment="Center"
+                                            Command="{Binding SplitButtonDefaultActionCommand}"
+                                            Content="Generate new name"
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:225]: Can you add strings for this?

File: Files/Dialogs/FilesystemOperationDialog.xaml [hunk 10]
```diff
@@ -38,174 +45,262 @@
...
+                                            Orientation="Horizontal"
+                                            Spacing="8"
+                                            Visibility="{Binding ShowUndoButton}">
+
+                                            <!--  Undo button  -->
+                                            <HyperlinkButton>
+                                                <HyperlinkButton.Content>
+                                                    <StackPanel Orientation="Horizontal" Spacing="8">
+                                                        <FontIcon FontSize="12" Glyph="&#xE10E;" />
+                                                        <TextBlock Text="Undo" />
```
Comment (yaichenbaum) [inline Files/Dialogs/FilesystemOperationDialog.xaml:258]: Can you add strings for this?

**Review by yaichenbaum — DISMISSED** [review]

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**